tensorflow/core/grappler/clusters/utils.cc

```diff
@@ -160,13 +160,13 @@ DeviceProperties GetLocalDMLInfo(int device_id) {
 #if TENSORFLOW_USE_DIRECTML
   auto adapters = EnumerateAdapters();
 
-  // When the D3D12 or DML device is removed after initially enumerating the
-  // devices, the adapters list will be empty
-  if (adapters.empty()) {
+  // After a DML device gets removed, the device IDs in subsequent runs may be
+  // out of bounds since the list is smaller by 1 element
+  if (device_id >= adapters.size()) {
     return device;
```

Comment:
How do we get into this state in the first place? The list of adapters is enumerated at startup, no?